pkg/querier/queryrange/step_align.go

```diff
@@ -6,17 +6,24 @@ import (
 
 // StepAlignMiddleware aligns the start and end of request to the step to
 // improved the cacheability of the query results.
-var StepAlignMiddleware = MiddlewareFunc(func(next Handler) Handler {
-	return stepAlign{
-		next: next,
+func StepAlignMiddleware(maxStepAlignment int64) MiddlewareFunc {
+	return func(next Handler) Handler {
+		return stepAlign{
+			next:             next,
+			maxStepAlignment: maxStepAlignment,
+		}
 	}
-})
+}
 
 type stepAlign struct {
-	next Handler
+	next             Handler
+	maxStepAlignment int64
 }
 
 func (s stepAlign) Do(ctx context.Context, r Request) (Response, error) {
+	if s.maxStepAlignment > 0 && r.GetStep() > s.maxStepAlignment {
```

Comment:
The `maxStepAlignment` is configured in seconds while `GetStep` returns milliseconds which would make it not work as expected.

Reply:
I've fixed it